modules/fedimint-ln-client/src/db.rs

```diff
@@ -247,18 +269,67 @@ pub(crate) fn get_v3_migrated_state(
                 out_points: v0.out_points,
                 error_reason: "unknown error (database migration)".to_string(),
             };
-            let new_pay = LightningPayStateMachine {
+            let new_pay = LightningPayStateMachineOld {
                 common,
                 state: LightningPayStates::Refund(v1),
             };
-            let new_sm = LightningClientStateMachines::LightningPay(new_pay);
-            let bytes = new_sm.consensus_encode_to_vec();
-            Ok(Some((bytes, operation_id)))
+            let pay_bytes = new_pay.consensus_encode_to_vec();
+            let mut sm_bytes: Vec<u8> = Vec::new();
+            1_u16.consensus_encode(&mut sm_bytes).unwrap();
+            let len = u16::try_from(pay_bytes.len()).unwrap();
+            len.consensus_encode(&mut sm_bytes).unwrap();
+            sm_bytes.extend_from_slice(&pay_bytes);
+            Ok(Some((sm_bytes, operation_id)))
         }
         _ => Ok(None),
     }
 }
 
+/// Adds the `amount` field to `LightningPayCommon`
+pub(crate) fn get_v4_migrated_state(
+    operation_id: OperationId,
+    cursor: &mut Cursor<&[u8]>,
+) -> anyhow::Result<Option<(Vec<u8>, OperationId)>> {
+    #[derive(Debug, Clone, Eq, PartialEq, Hash, Decodable, Encodable)]
+    pub struct LightningPayCommonOld {
+        pub operation_id: OperationId,
+        pub federation_id: FederationId,
+        pub contract: OutgoingContractData,
+        pub gateway_fee: Amount,
+        pub preimage_auth: sha256::Hash,
+        pub invoice: Bolt11Invoice,
+    }
+
+    let decoders = ModuleDecoderRegistry::default();
+    let ln_sm_variant = u16::consensus_decode(cursor, &decoders)?;
+
+    // If the state machine is not a pay state machine, return None
+    if ln_sm_variant != 1 {
+        return Ok(None);
+    }
+
+    let _ln_sm_len = u16::consensus_decode(cursor, &decoders)?;
+    let common = LightningPayCommonOld::consensus_decode(cursor, &decoders)?;
+    let state = LightningPayStates::consensus_decode(cursor, &decoders)?;
+
+    let new_common = LightningPayCommon {
+        operation_id: common.operation_id,
+        federation_id: common.federation_id,
+        contract: common.contract,
+        gateway_fee: common.gateway_fee,
+        preimage_auth: common.preimage_auth,
+        invoice: common.invoice,
+        amount: None,
+    };
+
+    let new_sm = LightningClientStateMachines::LightningPay(LightningPayStateMachine {
+        common: new_common,
+        state,
+    });
+    let bytes = new_sm.consensus_encode_to_vec();
```

Comment:
nit: Could use the logic from `get_v3_migrated_state` so that if this needs to be modified again we don't need to go back and fix `get_v4_migrated_state`